### ISSUE TITLE
☂️ eslint-plugin-vitest

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#4656]

**Comment by shulaoda** [comment]
Does `vitest/no-commented-out-tests` seem to be duplicated with `jest/no-commented-out-tests`?

**Comment by Boshen** [comment]
We need to map jest to vitest rules in [tasks/lint_rules](https://github.com/oxc-project/oxc/tree/main/tasks/lint_rules), they have the same implementation.

**Comment by shulaoda** [comment]
`Justfile` seems to not have the command `just new-vitest-rule`. Should we add it or use `just new-jest-rule`? @Boshen

**Comment by mysteryven** [comment]
> `Justfile` seems to not have the command `just new-vitest-rule`. Should we add it or use `just new-jest-rule`? @Boshen

Add this is welcome.

**Comment by no-yan** [comment]
### Rules Implemented but Not Listed in the Table
Some rules are already implemented but not marked in the table. They are located in `src/rules/jest` because they share similar logic between Jest and Vitest.

The following files mention Vitest, suggesting they may also apply to Vitest:

<details><summary>List of files mentioning Vitest</summary>
<p>

```bash
> grep -rl "vitest"  src/rules/jest
./expect_expect.rs
./no_identical_title.rs
./no_conditional_in_test.rs
./valid_expect.rs
./consistent_test_it.rs
./no_disabled_tests.rs
./no_commented_out_tests.rs
./valid_describe_callback.rs
./no_focused_tests.rs
./no_alias_methods.rs
./no_restricted_jest_methods.rs
./no_test_prefixes.rs
./no_conditional_expect.rs
./prefer_hooks_in_order.rs

``` 

</p>
</details> 

### Unable to Register Rules with the Same Name Across Different Plugins

Some rules have the same name as those in the Jest plugin, which prevents them from being registered in Vitest. This is because the macro cannot differentiate identical rule names between plugins.

For more details, see: [#6520](https://github.com/oxc-project/oxc/issues/6520)

**Comment by camchenry** [comment]
@no-yan I've opened a PR which will automatically keep this issue in sync with that list. Thanks for making a note of it. Hopefully that will make this more clear.

https://github.com/oxc-project/oxc/pull/6667

cc @DonIsaac @Boshen

**Comment by Afsoon** [comment]
Working in https://github.com/oxc-project/oxc/pull/16540 I have notice that `vitest/no-restricted-vi-methods` is implemented but the table show as not. As you can see

https://github.com/oxc-project/oxc/blob/d6d2bcd1c7b2524addab2394c2849ba809006567/crates/oxc_linter/src/utils/mod.rs#L51

Probably is due a name mistmatch. For Vitest the linter is named `no-restricted-vi-methods` but in jest is `no-restricted-jest-methods`. Following you can see how the jest rule have test case for the vitest context:

https://github.com/oxc-project/oxc/blob/d6d2bcd1c7b2524addab2394c2849ba809006567/crates/oxc_linter/src/rules/jest/no_restricted_jest_methods.rs#L198-L231

**Comment by connorshea** [comment]
Seems like we can do a few more "ports" relatively easily by adding support for the following existing jest rules as vitest rules? 

- [prefer-called-with](https://github.com/oxc-project/oxc/blob/c778cba9d5e828b7d021fe32297e798954e35575/crates/oxc_linter/src/rules/jest/prefer_called_with.rs): [vitest rule link](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/prefer-called-with.md)
- [no-test-return-statement](https://github.com/oxc-project/oxc/blob/397bcd5f0febc04b8bed16f0a5a8dc80ca2e5469/crates/oxc_linter/src/rules/jest/no_test_return_statement.rs): [vitest rule link](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/no-test-return-statement.md)
- [no-large-snapshots](https://github.com/oxc-project/oxc/blob/008e67ae2e743d41bf07447240fe09659a478259/crates/oxc_linter/src/rules/jest/no_large_snapshots.rs): [vitest rule link](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/no-large-snapshots.md)
- [require-hook](https://github.com/oxc-project/oxc/blob/ff1c284995aa43386a0f2c7afb3d3ec81cab8ecc/crates/oxc_linter/src/rules/jest/require_hook.rs): [vitest rule link](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/require-hook.md)
- [prefer-to-contain](https://github.com/oxc-project/oxc/blob/a7dd0c7b7df251c8160960399b64518a2148a424/crates/oxc_linter/src/rules/jest/prefer_to_contain.rs): [vitest rule link](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/prefer-to-contain.md)

**Comment by Afsoon** [comment]
Prefer-to-contain also lacks of fix, it's something the [eslint-plugin-jest have](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-to-contain.md) and vitest have. Oxc can detect if it's fixable but it stops there.

**Comment by Afsoon** [comment]
Currently I'm working to finish the remaining rules, as far I have divide the missing rules as:

- Only apply to vitest
  - [valid-title](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/valid-title.md)
  - [consistent-test-filename](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/consistent-test-filename.md)

- Missing but compatible to vitest and jest
  - [prefer-snapshot-hint](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/prefer-snapshot-hint.md)
  - [prefer-expect-assertions](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/prefer-expect-assertions.md)

- Compatible Rules
The mentioned, in the previous message, on top of that `prefer-to-contain` needs to be added the fix.

My main motivation is for my personal side projects I need some of this rules to remove eslint completly. I could have decided implemented only my rules, but I find more fair to complete all of missing rules.

**Comment by ryoppippi** [comment]
i also need vitest/no-importing-vitest-globals
https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/no-importing-vitest-globals.md

**Comment by connorshea** [comment]
I feel like I have kicked a puppy by fixing the plugin name and getting the rules list updated with all the new rules from the last year.

<img width="536" height="207" alt="Image" src="https://github.com/user-attachments/assets/e8d71e56-c99a-4049-ae86-9eed1c5241ed" />

**Comment by inglec-arista** [comment]
It seems to me that [vitest/prefer-vi-mocked](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/src/rules/prefer-vi-mocked.ts) and [unbound-method](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/src/rules/unbound-method.ts) do **not** actually require type information (contrary to the rule docs).
I reported some of these inconsistencies under https://github.com/vitest-dev/eslint-plugin-vitest/issues/902.

If this is correct, can we remove the "Not intending to implement" flag from these rules?

**Comment by connorshea** [comment]
> If this is correct, can we remove the "Not intending to implement" flag from these rules?

If that is true and there is no type information needed, then yeah I don't see any reason for us not to support them.

**Comment by G-Rath** [comment]
Note that the Vi and Jest versions of `unbound-method` do require type information but they are just extensions of the `@typescript-eslint` rule which [looks to already be supported](https://oxc.rs/docs/guide/usage/linter/rules/typescript/unbound-method).

Not sure if there's already support in place for extending rules, but I would imagine this makes it a lot easier to port these rules even though they require type information

**Comment by inglec-arista** [comment]
> > If this is correct, can we remove the "Not intending to implement" flag from these rules?
> 
> If that is true and there is no type information needed, then yeah I don't see any reason for us not to support them.

I was not correct about all of the rules mentioned, but `prefer-vi-mocked` has now been updated as not requiring type information.
https://github.com/vitest-dev/eslint-plugin-vitest/pull/905

**Comment by connorshea** [comment]
Opened https://github.com/oxc-project/oxc/pull/22724 to remove that vitest rule from the unsupported list
